### PR TITLE
Deprecate redux APIs

### PR DESCRIPTION
**Pull request by GerardasB**
## Changes

This PR deprecates APIs that are related to the redux store.

## Testing

N/A

### CODE REVIEW

File: ui/appui-react/src/appui-react/UiFramework.ts [hunk 1]
```diff
@@ -260,21 +238,27 @@ export class UiFramework {
...
    */
   public static async initialize(
     store: Store<any> | undefined,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    frameworkStateKey?: string
+  ): Promise<void>;
+  public static async initialize(
+    store?: Store<any> | undefined,
     frameworkStateKey?: string
   ): Promise<void> {
```
Comment (ignas-k) [inline ui/appui-react/src/appui-react/UiFramework.ts:260]: Why do we need two function overloads that are the identical?

**Review by ignas-k — APPROVED** [review]

File: ui/appui-react/src/appui-react/UiFramework.ts (hunk 1 above)
Comment (GerardasB) [reply ui/appui-react/src/appui-react/UiFramework.ts:260]: These are 2 signatures and 1 fn body. https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

File: ui/appui-react/src/appui-react/UiFramework.ts [hunk 2]
```diff
@@ -6,13 +6,12 @@
...
 
-// cSpell:ignore configurableui clientservices
-
 import type { Store } from "redux";
 import { Logger, ProcessDetector } from "@itwin/core-bentley";
 import type { TranslationOptions } from "@itwin/core-common";
 import type { IModelConnection, ViewState } from "@itwin/core-frontend";
-import { IModelApp, SnapMode } from "@itwin/core-frontend";
+import { SnapMode } from "@itwin/core-frontend";
+import { IModelApp } from "@itwin/core-frontend";
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/UiFramework.ts:14]: These specifiers are used, looks like there's no error

File: ui/appui-react/src/appui-react/statusfields/SelectionScope.tsx [hunk 3]
```diff
@@ -6,44 +6,69 @@
...
-  activeSelectionScope: string;
-  availableSelectionScopes: PresentationSelectionScope[];
+  /** Describes which selection scope is active. Uses redux store as a fallback. Defaults to `""`. */
+  activeScope?: string;
+  /** Describes available selection scoped. Uses redux store as a fallback. Defaults to `{ id: "element", label: "Element" }`. */
+  selectionScopes?: {
+    id: string;
+    label: string;
+  }[];
+  /** Describes available selection scoped. Uses redux store as a fallback. Defaults to `{ id: "element", label: "Element" }`. */
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/statusfields/SelectionScope.tsx]: Fixed.

File: ui/appui-react/src/appui-react/statusfields/SelectionScope.tsx [hunk 4]
```diff
@@ -6,44 +6,69 @@
...
 
-/** Defines properties supported by the SelectionScopeField Component.
- * @public
- */
 interface SelectionScopeFieldProps extends CommonProps {
-  activeSelectionScope: string;
-  availableSelectionScopes: PresentationSelectionScope[];
+  /** Describes which selection scope is active. Uses redux store as a fallback. Defaults to `""`. */
+  activeScope?: string;
+  /** Describes available selection scoped. Uses redux store as a fallback. Defaults to `{ id: "element", label: "Element" }`. */
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/statusfields/SelectionScope.tsx]: Fixed.

File: ui/appui-react/src/appui-react/redux/SessionState.ts [hunk 5]
```diff
@@ -64,28 +66,38 @@ export enum SessionStateActionId {
...
   defaultIModelViewportControlId: string | undefined;
+  /** @deprecated in 4.14.x. Not used by AppUI components. */
   defaultViewId: string | undefined;
+  /** @deprecated in 4.14.x. Use {@link UiFramework.getDefaultViewState} or {@link useActiveIModelConnection} instead. */
   defaultViewState: any;
+  /** @deprecated in 4.14.x. Use {@link UiFramework.getIModelConnection} or {@link useActiveIModelConnection} instead. */
   iModelConnection: any;
-  // eslint-disable-next-line deprecation/deprecation
-  cursorMenuData: CursorMenuData | undefined; // @deprecated in 4.11.x use {@link CursorMenuPayload} instead
+  /** @deprecated in 4.11.x use {@link CursorMenuPayload} instead */
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/redux/SessionState.ts]: Fixed.

File: ui/appui-react/src/appui-react/theme/ThemeManager.tsx [hunk 6]
```diff
@@ -182,15 +46,49 @@ class ThemeManagerComponent extends React.Component<ThemeManagerProps> {
  * and `ThemeProvider` theme will be set to `inherit`, in this case the application is
  * responsible for setting the theme by overriding iTwinUI css variables.
  *
- * This React component is Redux connected and should wrap [[ConfigurableUiContent]].
+ * This React component should wrap [[ConfigurableUiContent]].
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/theme/ThemeManager.tsx]: Fixed.

File: ui/appui-react/src/appui-react/theme/ThemeId.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  /** Will use iTwinUI wrapping `ThemeProvider` if provided, or default to 'Light' */
+  Inherit = "inherit",
+  /** Will use iTwinUI `light` theme with `highContrast` set to `true` */
+  HighContrastLight = "high-contrast-light",
+  /** Will use iTwinUI `dark` theme with `highContrast` set to `true` */
+  HighContrastDark = "high-contrast-dark",
+}
+
+/** Describe valid themes.
+ * See [[ThemeManager]] for more information.
```
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/theme/ThemeId.ts]: Fixed.

**Pull request merged**